route/route.go

```diff
@@ -203,114 +202,24 @@ func (r *Router) version(w http.ResponseWriter, req *http.Request) {
 func (r *Router) event(w http.ResponseWriter, req *http.Request) {
 	defer req.Body.Close()
 
-	// get out request ID for logging
-	reqID := req.Context().Value(types.RequestIDContextKey{})
-	debugLog := r.iopLogger.Debug().WithField("request_id", reqID)
-
-	reqBod, _ := ioutil.ReadAll(req.Body)
-	var trEv eventWithTraceID
-	// pull out just the trace ID for use in routing
-	err := unmarshal(req, bytes.NewReader(reqBod), &trEv)
+	bodyReader, err := r.getMaybeCompressedBody(req)
 	if err != nil {
-		debugLog.WithField("error", err.Error()).WithField("request.url", req.URL).WithField("json_body", string(reqBod)).Logf("error parsing json")
-		r.handlerReturnWithError(w, ErrJSONFailed, err)
-		return
-	}
-
-	// not part of a trace. send along upstream
-	if trEv.TraceID == "" {
-		r.Metrics.IncrementCounter(r.incomingOrPeer + "_router_event")
-		ev, err := r.requestToEvent(req, reqBod)
-		if err != nil {
-			r.handlerReturnWithError(w, ErrReqToEvent, err)
-			return
-		}
-		ev.Type = types.EventTypeEvent
-		ev.Target = types.TargetUpstream
-		debugLog.WithFields(map[string]interface{}{
-			"api_host": ev.APIHost,
-			"dataset":  ev.Dataset,
-		}).Logf("sending non-trace event")
-		r.UpstreamTransmission.EnqueueEvent(ev)
+		r.handlerReturnWithError(w, ErrPostBody, err)
 		return
 	}
 
-	// ok, we're a span. Figure out if we should handle locally or pass on to a
-	// peer
-	targetShard := r.Sharder.WhichShard(trEv.TraceID)
-	if !targetShard.Equals(r.Sharder.MyShard()) {
-		// it's not for us; send to the peer
-		r.Metrics.IncrementCounter(r.incomingOrPeer + "_router_peer")
-		ev, err := r.requestToEvent(req, reqBod)
-		if err != nil {
-			r.handlerReturnWithError(w, ErrReqToEvent, err)
-			return
-		}
-		ev.Type = types.EventTypeSpan
-		ev.Target = types.TargetPeer
-		ev.APIHost = targetShard.GetAddress()
-		debugLog.WithFields(map[string]interface{}{
-			"api_host": ev.APIHost,
-			"dataset":  ev.Dataset,
-			"trace_id": trEv.TraceID,
-			"peer":     targetShard.GetAddress(),
-		}).Logf("Sending span to my peer")
-		r.PeerTransmission.EnqueueEvent(ev)
-		return
-	}
-	// we're supposed to handle it
+	reqBod, _ := ioutil.ReadAll(bodyReader)
```

Comment:
in the batch endpoint we handle an error being returned from this call, should they both do the same thing?